addons/website_slides/models/slide_channel.py

```diff
@@ -87,11 +96,14 @@ def unlink(self):
             self.env['slide.slide.partner'].search(removed_slide_partner_domain).unlink()
         return super(ChannelUsersRelation, self).unlink()
 
-    def _set_as_completed(self):
-        """ Set record as completed and compute karma gains """
+    def _set_as_completed(self, completed=True):
+        """ Set record as completed and compute karma gains
+
+        :param completed: True if we will add karma points, False if we will remove karma points
```

Comment:
Well completed defines if we set the slide as completed or uncompleted first.
The fact that we gain or lose karma is a consequence of that.